### DISCUSSION TITLE
How to use `rehype-external-links` with `micromark`

### DISCUSSION THREAD

**Discussion by jespertheend** [rehypejs/rehype#132] [Q&A] [answered]
The [readme from `micromark`](https://github.com/micromark/micromark#extending-markdown) lists [`remark-external-links`](https://github.com/remarkjs/remark-external-links) as a potential extension for adding `_blank` to links, but it contains a deprecation notice which points me to [`rehype-external-links`](https://github.com/rehypejs/rehype-external-links). But I have no idea how to use this in combination with `micromark`.

Should I add it to `extensions`, `htmlExtensions`, call `rehypeExternalLinks()`? I have no idea.

**Comment by ChristianMurphy** [comment, marked as answer]
Usage would go through remark (which wraps micromark) and rehype.
For example https://github.com/rehypejs/rehype-external-links#use

```ts
import {unified} from 'unified'
import remarkParse from 'remark-parse'
import remarkRehype from 'remark-rehype'
import rehypeExternalLinks from 'rehype-external-links'
import rehypeStringify from 'rehype-stringify'

const file = await unified()
  .use(remarkParse)
  .use(remarkRehype)
  .use(rehypeExternalLinks, {rel: ['nofollow']})
  .use(rehypeStringify)
  .process('[rehype](https://github.com/rehypejs/rehype)')

console.log(String(file))
```

**Reply by jespertheend** [reply]
Ah, I'm guessing micromark did a bit too good of a job convincing me to use it directly instead of via unified then xD
Thanks! I'll try this out.

**Reply by jespertheend** [reply]
Ok wow I should have gone with unified from the start.
Took me a good couple of hours to figure out how to do what I want, but couldn't make it work with micromark.
I got a bit bummed out at the end of the day yesterday.
But using unified was so much easier. I even managed to write a small plugin to rewrite my links to the correct destination.
Documentation was also a lot more extensive.
